model/riscv_insts_vext_arith.sail

```diff
@@ -1714,174 +1714,73 @@ mapping clause assembly = WMVVTYPE(funct6, vm, vs2, vs1, vd)
-  let SEW = get_sew();
-  let LMUL_pow = get_lmul_pow();
-  let num_elem = get_num_elem(LMUL_pow, SEW);
-  let SEW_quart = SEW / 4;
-  let LMUL_pow_quart = LMUL_pow - 2;
-
-  if  illegal_variable_width(vd, vm, SEW_quart, LMUL_pow_quart) |
-      not(valid_reg_overlap(vs2, vd, LMUL_pow_quart, LMUL_pow))
-  then return Illegal_Instruction();
-
-  assert(SEW_quart >= 8);
-
-  let 'n = num_elem;
-  let 'm = SEW;
-  let 'o = SEW_quart;
-
-  let vm_val  : bits('n)             = read_vmask(num_elem, vm, zvreg);
-  let vd_val  : vector('n, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
-  let vs2_val : vector('n, bits('o)) = read_vreg(num_elem, SEW_quart, LMUL_pow_quart, vs2);
-
-  let (initial_result, mask) : (vector('n, bits('m)), bits('n)) = match init_masked_result(num_elem, SEW, LMUL_pow, vd_val, vm_val) {
-    Ok(v)   => v,
-    Err(()) => return Illegal_Instruction()
-  };
-  var result = initial_result;
-
-  foreach (i from 0 to (num_elem - 1)) {
-    if mask[i] == bitone then {
-      result[i] = match funct6 {
-        VEXT4_ZVF4 => zero_extend(vs2_val[i]),
-        VEXT4_SVF4 => sign_extend(vs2_val[i])
-      }
-    }
-  };
-
-  write_vreg(num_elem, SEW, LMUL_pow, vd, result);
-  set_vstart(zeros());
-  RETIRE_SUCCESS
-}
-
-mapping vext4type_mnemonic : vext4funct6 <-> string = {
-  VEXT4_ZVF4  <-> "vzext.vf4",
-  VEXT4_SVF4  <-> "vsext.vf4"
-}
-
-mapping clause assembly = VEXT4TYPE(funct6, vm, vs2, vd)
-  <-> vext4type_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ maybe_vmask(vm)
-
-/* ****************************** OPMVV (VXUNARY0) ******************************* */
-/* ******************* Vector Integer Extension (SEW/8 source) ******************* */
-union clause ast = VEXT8TYPE : (vext8funct6, bits(1), vregidx, vregidx)
-
-mapping vext8_vs1 : vext8funct6 <-> bits(5) = {
+  VEXT4_SVF4  <-> 0b00101,
   VEXT8_ZVF8  <-> 0b00010,
-  VEXT8_SVF8  <-> 0b00011
+  VEXT8_SVF8  <-> 0b00011,
 }
 
-mapping clause encdec = VEXT8TYPE(funct6, vm, vs2, vd)
-  <-> 0b010010 @ vm @ encdec_vreg(vs2) @ vext8_vs1(funct6) @ 0b010 @ encdec_vreg(vd) @ 0b1010111
+mapping clause encdec = VEXTTYPE(funct6, vm, vs2, vd)
+  <-> 0b010010 @ vm @ encdec_vreg(vs2) @ vext_vs1(funct6) @ 0b010 @ encdec_vreg(vd) @ 0b1010111
   when currentlyEnabled(Ext_V)
 
-function clause execute(VEXT8TYPE(funct6, vm, vs2, vd)) = {
+function clause execute(VEXTTYPE(funct6, vm, vs2, vd)) = {
   let SEW = get_sew();
   let LMUL_pow = get_lmul_pow();
   let num_elem = get_num_elem(LMUL_pow, SEW);
-  let SEW_eighth = SEW / 8;
-  let LMUL_pow_eighth = LMUL_pow - 3;
-
-  if  illegal_variable_width(vd, vm, SEW_eighth, LMUL_pow_eighth) |
-      not(valid_reg_overlap(vs2, vd, LMUL_pow_eighth, LMUL_pow))
+  let div_num : {2, 4, 8} = match funct6 {
+    VEXT2_ZVF2  =>  2,
+    VEXT2_SVF2  =>  2,
+    VEXT4_ZVF4  =>  4,
+    VEXT4_SVF4  =>  4,
+    VEXT8_ZVF8  =>  8,
+    VEXT8_SVF8  =>  8,
+  };
+  let SEW_div = SEW / div_num;
```

Comment:
Can you just do `let SEW_div = SEW / (2 ^ sub_num);` and get rid of `div_num`?

Also I'd maybe call `sub_num` something like `SEW_frac_pow`?